scipy/stats/tests/test_distributions.py

```diff
@@ -6336,6 +6336,23 @@ def test_entropy(self):
                         stats.norm.entropy(loc=1.0, scale=2.5), rtol=0.05)
 
 
+def test_histogram_non_uniform():
+    # Tests rv_histogram works even for non-uniform bin widths
+    counts, bins = ([1, 1], [0, 1, 1001])
+    dist = stats.rv_histogram((counts, bins), density=False)
+    np.testing.assert_allclose(
+        dist.pdf([0.5, 200]),
+        np.array([0.5, 0.0005]))
+    assert dist.median() == 1
+    # Omitting density produces a warning for non-uniform bins...
+    with assert_warns(RuntimeWarning):
+        dist = stats.rv_histogram((counts, bins))
+    # ... but not for uniform bins
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
```

Comment:
Why do we need the filter here?

Reply:
My idea was to test that the warning is not simply triggered all the time. So the filter temporarily promoted the warning to an exception, making the test fail if the warning occurs when it shouldn't. But maybe you would catch warnings that occur during tests already in other ways; I know very little about scipy's build/test process.

Reply:
Yup, by default warnings are treated as a test failures unless they are suppressed or part of the test.